src/main/java/seedu/foodrem/views/ItemView.java

```diff
@@ -0,0 +1,115 @@
+package seedu.foodrem.views;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.control.Separator;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.VBox;
+import javafx.scene.text.TextAlignment;
+import seedu.foodrem.model.item.Item;
+import seedu.foodrem.model.tag.Tag;
+
+/**
+ * A view of an {@code Item}. This can be displayed.
+ * @author Richard Dominick
+ */
+public class ItemView {
+    private static final double SPACING_UNIT = 8;
+
+    private ItemView() {} // Prevents instantiation
+
+    /**
+     * Creates a new detailed view of the given item.
+     * @param item the item to be displayed.
+     * @return the node to be displayed in the UI.
+     */
+    public static Node from(Item item) {
+        final Label name = new Label(item.getName().toString());
+        name.getStyleClass().add("item-detail-name");
+        name.prefWidth(Double.MAX_VALUE);
+
+        // Name and tags at the top left
+        final List<Node> tagsList = new ArrayList<>();
+        tagsList.add(new Label("Tags: "));
+        item.getTagSet().stream().sorted(Comparator.comparing(Tag::getName))
+                .forEach(tag -> tagsList.add(buildTagNodeFrom(tag.getName())));
+        if (tagsList.size() == 1) {
+            tagsList.add(new Label("-"));
+        }
```

Comment:
This logic can be abstracted out as we need it in other places like listing all tags

Reply:
Will resolve this in a future PR

Reply:
I think this is fine for now. We abstract out as necessary. But good point, this should be kept in mind every time we are adding something new: can stuff be abstracted out instead? That way we don't overcomplicate things at the start. YAGNI goes both ways, we may not need the abstracted method at all.